pkg/cache/cache_from_test.go

```diff
@@ -0,0 +1,221 @@
+	return m.imageCtrl
+}
+
+func (*mockRegistry) HasGlobalPushAccess() (bool, error) {
+	return true, nil
+}
+
+type mockRegistryWithError struct {
+	imageCtrl ImageCtrlInterface
+}
+
+func (m *mockRegistryWithError) GetImageCtrl() ImageCtrlInterface {
+	return m.imageCtrl
+}
+
+func (*mockRegistryWithError) HasGlobalPushAccess() (bool, error) {
+	return false, assert.AnError
+}
+
+type mockImageCtrl struct{}
+
+func (*mockImageCtrl) GetRegistryAndRepo(_ string) (string, string) {
+	return "registry", "test-account/test-image:x.y.z"
+}
+
+func (*mockImageCtrl) GetRepoNameAndTag(_ string) (string, string) {
+	return "test-account/test-image", "x.y.z"
+}
+
+func Test_AddDefaultPullCache(t *testing.T) {
+	imageCtrl := &mockImageCtrl{}
+	reg := &mockRegistry{imageCtrl: imageCtrl}
+
+	tests := []struct {
+		name     string
+		image    string
+		cf       CacheFrom
+		expected CacheFrom
+	}{
+		{
+			name:     "already with cache image",
+			image:    "test-account/test-image:1.0.0",
+			cf:       CacheFrom{"okteto.global/test-account/test-image:cache", "okteto.dev/test-account/test-image:cache"},
+			expected: CacheFrom{"okteto.global/test-account/test-image:cache", "okteto.dev/test-account/test-image:cache"},
+		},
+		{
+			name:     "without cache image",
+			image:    "test-account/test-image:1.0.0",
+			cf:       CacheFrom{},
+			expected: CacheFrom{"okteto.global/test-account/test-image:cache", "okteto.dev/test-account/test-image:cache"},
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			tt.cf.AddDefaultPullCache(reg, tt.image)
+			assert.Equal(t, tt.expected, tt.cf)
+		})
+	}
+}
+
+func Test_AddDefaultPullCache_WithError(t *testing.T) {
+	image := "test-account/test-image:x.y.z"
+	cf := CacheFrom{}
+	expected := CacheFrom{"okteto.dev/test-account/test-image:cache"}
+
+	imageCtrl := &mockImageCtrl{}
+	registry := &mockRegistryWithError{imageCtrl: imageCtrl}
+
+	cf.AddDefaultPullCache(registry, image)
+
+	assert.Equal(t, cf, expected)
+}
+
+func Test_hasCacheFromImage(t *testing.T) {
+	tests := []struct {
+		name     string
+		input    string
+		cf       CacheFrom
+		expected bool
+	}{
+		{
+			name:     "not found",
+			input:    "test-registry/test-image:cache",
+			cf:       CacheFrom{"test-registry/test-image:test-tag"},
+			expected: false,
+		},
+		{
+			name:     "found",
+			input:    "test-registry/test-image:cache",
+			cf:       CacheFrom{"test-registry/test-image:cache"},
+			expected: true,
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			result := tt.cf.hasCacheFromImage(tt.input)
+			assert.Equal(t, tt.expected, result)
+		})
+	}
+}
+
+func Test_UnmarshalYAML(t *testing.T) {
+	tests := []struct {
+		name     string
+		data     []byte
+		expected interface{}
+	}{
+		{
+			name:     "empty",
+			data:     []byte("[]"),
+			expected: CacheFrom{},
+		},
+		{
+			name:     "one item",
+			data:     []byte(`["test-registry/test-image:cache"]`),
```

Comment:
```suggestion
			data:     []byte(`"test-registry/test-image:cache"`),
```
If we send a list we are using the same test case as the last one